multi_partner_learning.py

```diff
@@ -85,26 +87,26 @@ def compute_test_score_for_single_partner(self, partner):
 
         # Train model
         logger.info("   Training model...")
-        history = model.fit(
-            partner.x_train,
-            partner.y_train,
-            batch_size=partner.batch_size,
-            epochs=self.epoch_count,
-            verbose=0,
-            validation_data=self.val_data,
-            callbacks=cb,
-        )
-
-        # Reference the testset according to the scenario configuration
-        x_test, y_test = self.test_data
+        if isinstance(model, type(LogisticRegression())):
+            history = model.fit(partner.x_train, partner.y_train)
+        else:
+            history = model.fit(
+                partner.x_train,
+                partner.y_train,
+                batch_size=partner.batch_size,
+                epochs=self.epoch_count,
+                verbose=0,
+                validation_data=self.val_data,
+                callbacks=cb,
+            )
```

Comment:
It would be best to have this conditional statement in a dedicated fit function, to keep `compute_test_score_for_single_partner()` independent from the model type.

Reply:
Very nice suggestion. It simplifies the logic a lot, the `fit_model()` function was almost already able to handle single partner scenarii.